src/zkevm_specs/evm/execution/end_block.py

```diff
@@ -68,6 +81,16 @@ def end_block(instruction: Instruction):
             instruction.constrain_equal(
                 instruction.call_context_lookup(CallContextFieldTag.TxId), total_txs
             )
+            # 4. Verify that CumulativeGasUsed does not exceed the block gas limit.
+            gas_limit = instruction.block_context_lookup(BlockContextFieldTag.GasLimit)
+            cumulative_gas = instruction.tx_receipt_read(
+                total_txs,
+                TxReceiptFieldTag.CumulativeGasUsed,
+                # +1 to rw_counter to skip lookup to PostStateOrStatus
+                instruction.curr.rw_counter + instruction.rw_counter_offset + 1,
```

Comment:
I don't follow this value.  I think we can pass `None` here so that the logic is of `Instruction.rw_lookup` applied:
```python
            rw_counter = self.curr.rw_counter + self.rw_counter_offset
            self.rw_counter_offset += 1
```

The EndBlock performs a list of lookups to rw_table which correspond to some new rw operations (that will have contiguous rw_counters).